### PR TITLE
Add code search notebook

### PR DESCRIPTION
**Pull request by kacperlukawski**
This is a notebook created for the purposes of a code search tutorial.

See: https://github.com/qdrant/landing_page/pull/661

### CODE REVIEW

File: code-search/code-search.ipynb [hunk 1]
```diff
@@ -0,0 +1,609 @@
...
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "iD8Fdu1SWw_7"
+      },
+      "outputs": [],
+      "source": [
+        "!wget https://storage.googleapis.com/tutorial-attachments/code-search/structures.jsonl"
```
Comment (NirantK) [inline code-search/code-search.ipynb:42]: Would you like to use Huggingface Datasets instead? It's as programmatic as wget, with the advantage of us getting a sense of how many people download it?

File: code-search/code-search.ipynb [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "2DFkoAUSiW0Z"
+      },
+      "outputs": [],
+      "source": [
+        "from sentence_transformers import SentenceTransformer\n",
```
Comment (NirantK) [inline code-search/code-search.ipynb:206]: Would really appreciate it if we used FastEmbed here — FastEmbed has this sentence transformer model as well, and it's a tad bit faster too!

File: code-search/code-search.ipynb [hunk 3]
```diff
@@ -0,0 +1,609 @@
...
+      "execution_count": null,
+      "metadata": {
+        "id": "JcPlZxZmi4Ag"
+      },
+      "outputs": [],
+      "source": [
+        "code_model = SentenceTransformer(\n",
+        "    \"jinaai/jina-embeddings-v2-base-code\",\n",
+        "    token=HF_TOKEN,\n",
+        "    trust_remote_code=True\n",
```
Comment (NirantK) [inline code-search/code-search.ipynb:284]: Do you want to explain a bit here what `trust_remote_code=True` is doing? That might be new to quite a few folks?

File: code-search/code-search.ipynb [hunk 4]
```diff
@@ -0,0 +1,609 @@
...
+        "    \"qdrant-sources\",\n",
+        "    vectors_config={\n",
+        "        \"text\": models.VectorParams(\n",
+        "            size=nlp_embeddings.shape[1],\n",
+        "            distance=models.Distance.COSINE,\n",
+        "        ),\n",
+        "        \"code\": models.VectorParams(\n",
+        "            size=code_embeddings.shape[1],\n",
+        "            distance=models.Distance.COSINE,\n",
+        "        ),\n",
```
Comment (NirantK) [inline code-search/code-search.ipynb:335]: I'd love it if we can mention / call out that this is what Qdrant calls Named Vectors

File: code-search/code-search.ipynb [hunk 5]
```diff
@@ -0,0 +1,609 @@
...
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "results = client.search_groups(\n",
+        "    \"qdrant-sources\",\n",
+        "    query_vector=(\n",
+        "        \"code\", code_model.encode(query).tolist()\n",
+        "    ),\n",
+        "    group_by=\"context.module\",\n",
```
Comment (NirantK) [inline code-search/code-search.ipynb:562]: This looks excellent! 

Do you want to show inline how the output might look like for this? E.g. will it be two lists or one? Are groups just lists and so on?

**Review by NirantK — APPROVED** [review]
I've not run the code, since it mentions a GPU accelerator. That said, looks good to me!

File: code-search/code-search.ipynb (hunk 3 above)
Comment (kacperlukawski) [reply code-search/code-search.ipynb:284]: It is actually a part of the tutorial itself, but that might be a good idea to describe it quickly here. Especially due to the recent news about HF hosting some malicious models.

File: code-search/code-search.ipynb (hunk 2 above)
Comment (kacperlukawski) [reply code-search/code-search.ipynb:206]: I initially used FastEmbed, but I wanted unified access to both models, and Jina code embeddings are not supported, right?

File: code-search/code-search.ipynb (hunk 5 above)
Comment (kacperlukawski) [reply code-search/code-search.ipynb:562]: It actually makes sense to add all the outputs. I'll run the notebook and upload a version with outputs.

**Comment by kacperlukawski** [comment]
@NirantK I would stick with the

**Pull request merged**